src/components/DeskCard/DeskCard.tsx

```diff
@@ -18,12 +17,24 @@ const FeaturesBox = styled(Box)`
     gap: ${({ theme }) => theme.spacing(1)};
 `;
 
+const HeaderBox = styled(Box)`
+    display: flex;
+    justify-content: space-between;
+    align-items: center;
+    margin-bottom: ${({ theme }) => theme.spacing(2)};
+`;
 
+type DeskCardProps = Desk & {
+  onReserveClick: (deskId: string) => void
+}
 
-export const DeskCard = (props: Desk) => {
+export const DeskCard = (props: DeskCardProps) => {
   return (
     <CardStyled>
-      <DeskNameTypography variant="h6">{props.name}</DeskNameTypography>
+      <HeaderBox>
+        <DeskNameTypography variant="h6">{props.name}</DeskNameTypography>
+        {props.isAvailable && <Button variant='contained' onClick={() => props.onReserveClick(props.id)}>reserve</Button>}
```

Comment:
Let's also move 'not available' from the bottom here, displaying either it or the button depending on availability. From UX perspective it is better to have it at the same place.